slidez-core/src/main/java/com/binarystudio/academy/slidez/domain/sessionEvent/mapper/jsonb/JsonbType.java

```diff
@@ -0,0 +1,125 @@
+package com.binarystudio.academy.slidez.domain.sessionEvent.mapper.jsonb;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
+import java.io.StringWriter;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Types;
+
+import com.binarystudio.academy.slidez.domain.session.event.DomainEvent;
+import lombok.Data;
+import org.hibernate.HibernateException;
+import org.hibernate.engine.spi.SharedSessionContractImplementor;
+import org.hibernate.usertype.UserType;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+@Data
+public class JsonbType implements UserType {
+
+	@Override
+	public int[] sqlTypes() {
+		return new int[] { Types.OTHER };
+	}
+
+	@Override
+	public Class<DomainEvent> returnedClass() {
+		return DomainEvent.class;
+	}
+
+	@Override
+	public Object nullSafeGet(final ResultSet rs, final String[] names, final SharedSessionContractImplementor session,
+			final Object owner) throws HibernateException, SQLException {
+		final String cellContent = rs.getString(names[0]);
+		if (cellContent == null) {
+			return null;
+		}
+		try {
+			final ObjectMapper mapper = new ObjectMapper();
+			mapper.findAndRegisterModules();
+			return mapper.readValue(cellContent.getBytes("UTF-8"), returnedClass());
+		}
+		catch (final Exception ex) {
+			throw new RuntimeException("Failed to convert String to Invoice: " + ex.getMessage(), ex);
+		}
+	}
+
+	@Override
+	public void nullSafeSet(final PreparedStatement ps, final Object value, final int idx,
+			final SharedSessionContractImplementor session) throws HibernateException, SQLException {
+		if (value == null) {
+			ps.setNull(idx, Types.OTHER);
+			return;
+		}
+		try {
+			final ObjectMapper mapper = new ObjectMapper();
+			mapper.findAndRegisterModules();
+			final StringWriter w = new StringWriter();
+			mapper.writeValue(w, value);
+			w.flush();
+			ps.setObject(idx, w.toString(), Types.OTHER);
+		}
+		catch (final Exception ex) {
+			throw new RuntimeException("Failed to convert Invoice to String: " + ex.getMessage(), ex);
+		}
+	}
+
+	@Override
+	public Object deepCopy(final Object value) throws HibernateException {
+		try {
+			// use serialization to create a deep copy
+			ByteArrayOutputStream bos = new ByteArrayOutputStream();
```

Comment:
bos and oos will not be closed in case of the exception. Ideally, we should use https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

Reply:
fix this, thanks)